### ISSUE TITLE
Dependency and python updates

### ISSUE THREAD

**Issue by JacobHayes** [sokrypton/ColabFold#357]
Python 3.7 is end-of-life [in 5 months](https://endoflife.date/python) and 3.11 is the latest release, so the current supported range of 3.7 and 3.8 is aging quickly. Additionally, a few of the package dependencies are aging too and conflicts only get trickier to deal with over time. The main ones I noticed are:
- `matplotlib==3.1.3`: I think the matplotlib pin can be removed now (the reason for the pin [seems to be gone now](https://github.com/facebook/prophet/issues/1691#issuecomment-1397802599))
- `absl-py^0.13.0`: this prevents v1 from being used, which was [just a major bump to drop support for Python 2.7, 3.4, 3.5](https://github.com/abseil/abseil-py/blob/main/CHANGELOG.md#100-2021-11-09)
- `tensorflow-cpu<2.8.0`: this was pinned the same day 2.8.0 came out, not sure if new versions have the same issue, but seems likely they would have fixed it. The original issues seems very similar to https://github.com/tensorflow/tensorflow/issues/53038, which was an issue w/ numpy being outdated (guessing some compiled code mismatch?)
- `importlib-metadata>=4.8.2`: pretty minor, 5 and 6 were recently bumped and seem minor

I'm currently trying to integrate with a 3.10 codebase, so can open a PR to try relaxing some of the versions. I'm not too familiar with things - should test coverage catch most things? 🤞

**Comment by JacobHayes** [comment]
I think keeping python 3.7 support is the hardest part of updating the deps (eg: the last pandas release to support 3.7 is 1.3 from Dec 2021). After relaxing the deps but keeping `python>=3.7.1`, `poetry lock` took a few minutes. Changing to `python>=3.8` locked in 7s.

Would dropping 3.7 be an option? Guessing this is still used in quite a lot of 3.7 envs

**Comment by milot-mirdita** [comment]
I don't think we want to drop 3.7 quite yet. We restricted 3.9 since we had the following numpy error and there was no clear way to fix it:
```
RuntimeError: module compiled against API version 0x10 but this version of numpy is 0xe
```
since then we didn't bother to try/enable it again.

Tensorflow 2.8.0 was a bigger issue, however I don't remember what was wrong anymore.

**Comment by JacobHayes** [comment]
It looks like the tensorflow issue was https://github.com/sokrypton/ColabFold/issues/155, which also noted the numpy issue, and was fixed [here](https://github.com/tensorflow/tensorflow/issues/53038) with a numpy upgrade (ie: you could either downgrade tensorflow or upgrade numpy).

I'd guess both were the same root cause - tensorflow compiled with a newer version of numpy (which is why the upgrade fixed it). The [oldest-supported-numpy](https://pypi.org/project/oldest-supported-numpy/) package was made for that issue and lets you compiles against the oldest numpy version supported for that py version - doesn't seem tensorflow quite uses that.
